### PR TITLE
ROC 5005 investigation when claim generation fails after successful payment 

### PR DESCRIPTION
**Pull request by tharacka**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-5005

### Change description

Applying app-insights to the pay screen to get further information to diagnose the issue

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/claim/routes/pay.ts [hunk 1]
```diff
@@ -141,8 +147,21 @@ export default express.Router()
       draft.document.claimant.payment = payment
       await new DraftService().save(draft, user.bearerToken)
 
+      trackCustomEvent(`Pre payment for externalId: ${externalId}`,{
```
Comment (ellie-harrison) [inline src/main/features/claim/routes/pay.ts]: I dont think there's any value in tracking this event, we only really need to know when this fails.

File: src/main/features/claim/routes/pay.ts [hunk 2]
```diff
@@ -165,6 +184,12 @@ export default express.Router()
 
       await new DraftService().save(draft, user.bearerToken)
 
+      trackCustomEvent(`Post payment for externalId: ${externalId}`,{
```
Comment (ellie-harrison) [inline src/main/features/claim/routes/pay.ts]: Same here, I think there's only value in tracking the failure

File: src/main/app/logging/customEventTracker.ts [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
+export function trackCustomEvent (eventName: string, trackingProperties: {}) {
+  try {
+    if (appInsights.defaultClient) {
+      appInsights.defaultClient.trackEvent({
+        name: eventName,
+        properties: trackingProperties
+      })
+    }
+  } catch (err) {
+    return
```
Comment (ellie-harrison) [inline src/main/app/logging/customEventTracker.ts]: Should we be doing something with this error than just returning?

File: src/main/features/claim/routes/pay.ts (hunk 1 above)
Comment (tharacka) [reply src/main/features/claim/routes/pay.ts]: The reason why I originally put them in is to ensure that the microservices being used in pay.ts were behaving. Also, I also wanted to make sure that the RC emails being generated reflected the correct status. However, for the time being I see your point and have removed them

File: src/main/app/logging/customEventTracker.ts (hunk 3 above)
Comment (tharacka) [reply src/main/app/logging/customEventTracker.ts]: As above, have been removed

File: src/main/features/claim/routes/pay.ts (hunk 2 above)
Comment (tharacka) [reply src/main/features/claim/routes/pay.ts]: As above

**Pull request merged**